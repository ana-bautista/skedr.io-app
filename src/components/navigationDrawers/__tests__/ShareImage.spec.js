import { shallowMount, mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Comp from '../ShareImage.vue'

Vue.use(Vuex)
Vue.use(Vuetify)

const createStore = ({ photos, isEmpty = false, hasBackup = false, actions = {} }) =>
  new Vuex.Store({
    actions,
    modules: {
      pool: {
        namespaced: true,
        state: { photos },
        getters: { isEmpty: () => isEmpty, hasBackup: () => hasBackup }
      }
    }
  })

describe('ShareImage', () => {
  it('has one image', () => {
    const wrapper = shallowMount(Comp, {
      store: createStore({ photos: [{ photoId: 'photoId' }] })
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('has no images and has backup', () => {
    const wrapper = shallowMount(Comp, {
      store: createStore({ photos: [{ photoId: 'photoId' }], isEmpty: true, hasBackup: true })
    })

    expect(wrapper).toMatchSnapshot()
  })

  describe('Actions', () => {
    it('clicks share', () => {
      const actions = { 'sharedPool/share': jest.fn() }
      const photos = [{ photoId: 'photoId' }]
      const wrapper = mount(Comp, {
        store: createStore({
          photos,
          hasBackup: false,
          actions
        }),
        stubs: ['share-dialog']
      })
      const button = wrapper.findAll('button')
      button.at(0).trigger('click')
      expect(actions['sharedPool/share']).toHaveBeenCalled()
      //expect(actions['sharedPool/share']).toHaveBeenNthCalledWith(1, { pool: photos })
    })
    it('clicks clearPool', () => {
      const actions = { 'pool/clearPool': jest.fn() }
      const wrapper = mount(Comp, {
        store: createStore({
          photos: [{ photoId: 'photoId' }],
          hasBackup: false,
          actions
        }),
        stubs: ['share-dialog']
      })
      const button = wrapper.findAll('button')
      button.at(1).trigger('click')
      expect(actions['pool/clearPool']).toHaveBeenCalled()
    })
    it('clicks restoreBackup', () => {
      const actions = { 'pool/restoreBackup': jest.fn() }
      const wrapper = mount(Comp, {
        store: createStore({
          photos: [{ photoId: 'photoId' }],
          hasBackup: true,
          actions
        }),
        stubs: ['share-dialog']
      })
      const button = wrapper.findAll('button')
      button.at(1).trigger('click')
      expect(actions['pool/restoreBackup']).toHaveBeenCalled()
    })
  })
})
