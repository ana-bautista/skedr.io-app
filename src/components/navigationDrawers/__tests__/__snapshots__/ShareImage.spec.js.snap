// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareImage has no images and has backup 1`] = `
<v-navigation-drawer-stub fixed="true" app="true" clipped="true" height="100%" minivariantwidth="80" mobilebreakpoint="1264" right="true" width="300">
  <v-container-stub tag="div" fluid="" grid-list-md="" pa-0="">
    <v-layout-stub tag="div" py-2="" px-3="" fluid="">
      <v-flex-stub tag="div" xs6="">
        <v-tooltip-stub opendelay="200" closedelay="200" lazy="true" contentclass="" bottom="true" fixed="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" debounce="0" tag="span">
          <v-btn-stub color="success" activeclass="v-btn--active" disabled="true" tag="button" block="true" outline="true" small="true" type="button">
            <v-icon-stub>share</v-icon-stub><span>&nbsp;Sked</span>
          </v-btn-stub> <span>Select 1 or more photos</span>
        </v-tooltip-stub>
        <share-dialog-stub></share-dialog-stub>
      </v-flex-stub>
      <v-flex-stub tag="div" xs6="">
        <v-tooltip-stub opendelay="200" closedelay="200" lazy="true" contentclass="" bottom="true" fixed="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" debounce="0" tag="span">
          <v-btn-stub color="accent" activeclass="v-btn--active" tag="button" block="true" outline="true" small="true" type="button">
            <v-icon-stub>restore_from_trash</v-icon-stub>&nbsp;Restore
          </v-btn-stub> <span>Restore previous pool</span>
        </v-tooltip-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-list-stub><span name="list"><!----> <share-image-list-stub photo="[object Object]"></share-image-list-stub></span></v-list-stub>
  </v-container-stub>
</v-navigation-drawer-stub>
`;

exports[`ShareImage has one image 1`] = `
<v-navigation-drawer-stub fixed="true" app="true" clipped="true" height="100%" minivariantwidth="80" mobilebreakpoint="1264" right="true" width="300">
  <v-container-stub tag="div" fluid="" grid-list-md="" pa-0="">
    <v-layout-stub tag="div" py-2="" px-3="" fluid="">
      <v-flex-stub tag="div" xs6="">
        <v-tooltip-stub opendelay="200" closedelay="200" lazy="true" contentclass="" bottom="true" fixed="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" debounce="0" tag="span">
          <v-btn-stub color="success" activeclass="v-btn--active" tag="button" block="true" outline="true" small="true" type="button">
            <v-icon-stub>share</v-icon-stub><span>&nbsp;Sked</span>
          </v-btn-stub> <span>Share 1 images</span>
        </v-tooltip-stub>
        <share-dialog-stub></share-dialog-stub>
      </v-flex-stub>
      <v-flex-stub tag="div" xs6="">
        <v-tooltip-stub opendelay="200" closedelay="200" lazy="true" contentclass="" bottom="true" fixed="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" debounce="0" tag="span">
          <v-btn-stub color="error" activeclass="v-btn--active" tag="button" block="true" outline="true" small="true" type="button">
            <v-icon-stub>delete</v-icon-stub>&nbsp;Clear
          </v-btn-stub> <span>Clear pool</span>
        </v-tooltip-stub>
      </v-flex-stub>
    </v-layout-stub>
    <v-list-stub><span name="list"><!----> <share-image-list-stub photo="[object Object]"></share-image-list-stub></span></v-list-stub>
  </v-container-stub>
</v-navigation-drawer-stub>
`;
